### PR TITLE
[fix](io): use try with resource make io stream close automatically to avoid resource leak

### PR DESCRIPTION
**Pull request by xutaoustc**
Some code are scanned to have potential security and performance problems:
This pr fix the problem that some iostream are not closed after using.
Use try with resource make io stream close automatically to avoid resource leak

### CODE REVIEW

File: fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java [hunk 1]
```diff
@@ -44,18 +44,16 @@ public static boolean copy(Writable orig, Writable dest, Class c, int metaVersio
...
         metaContext.setThreadLocalInfo();
 
-        FastByteArrayOutputStream byteArrayOutputStream = new FastByteArrayOutputStream();
-        DataOutputStream out = new DataOutputStream(byteArrayOutputStream);
-        try {
+        try (FastByteArrayOutputStream byteArrayOutputStream = new FastByteArrayOutputStream();
+                DataOutputStream out = new DataOutputStream(byteArrayOutputStream)) {
             orig.write(out);
             out.flush();
             out.close();
```
Comment (yiguolei) [inline fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java:51]: Is this line useful?

File: fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java [hunk 2]
```diff
@@ -74,19 +72,17 @@ public static <T> T copy(Writable orig, Class<T> c, int metaVersion) {
...
```
Comment (yiguolei) [inline fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java:79]: and this line

File: fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java (hunk 1 above)
Comment (CalvinKirs) [reply fe/fe-common/src/main/java/org/apache/doris/common/io/DeepCopy.java:51]: don't need to explicitly invoke close

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**